common/bitvec/src/lib.rs

```diff
@@ -75,6 +79,29 @@ impl BitVec {
         let bucket_pos = pos as usize - (bucket * BUCKET_SIZE);
         (self.inner[bucket] & (0x01 << bucket_pos)) != 0
     }
+
+    // TODO(kostas): Remove after applying it to multi-sig.
+    #[allow(dead_code)]
+    /// Returns the number of set bits.
+    pub fn count_ones(&self) -> u32 {
+        self.inner.iter().map(|a|a.count_ones()).sum()
+    }
+
+    // TODO(kostas): Remove after applying it to multi-sig.
+    #[allow(dead_code)]
+    /// Returns the index of the last set bit or -1 if no bits are set.
+    pub fn last_set_bit(&self) -> i32 {
```

Comment:
as an alternate implementation, we could cache both `count_ones` and `last_bit_set` in struct members. The onus would move on deserialization impl to do these computations, but they would only need to be done once when reading from storage/network. If this is something you've already considered, might be good to mention the reasoning in the PR / commit message. If not, what are your thoughts?

Reply:
Good point, but because there will only be limited use cases requiring such a functionality and because we mostly need to compute this once after deserialization, we decided not to cache these numbers.

Reply:
Sounds good! Only request would be to add the trade-offs considered to the PR/commit message.

Reply:
Also, can we return `Option<u8>` rather than using the c-like API of returning -1? The impl can also be more concise. Example:
```rust
fn find_last() -> Option<u8> {
    let x: Vec<u8> = vec![0x11, 0x0];
    x.iter().rev().enumerate().filter(|(_, byte)| byte != &&0u8).nth(0).map(|(i, byte)|
         (8 * (x.len() - i) as u8) - (byte.trailing_zeros() as u8) - 1u8
    )
}
```

Reply:
@kchalkias : Thoughts on using `.nth(0)` and `.map` instead of using the for  loop, and returning Option<u8> instead of i32?

Reply:
@bothra90 done, back to you!